interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: Parameters are used in creating and declaring a function. While arguments are used when invoking the function.
 
-  Researched answer:
+  Researched answer: Function parameters specify a single or list of variables when declaring a function. Function arguments are passing/assigning value to a declared function when invoked. 
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: getorDefault uses the key value and general purpose is a static method
 
-  Researched answer:
+  Researched answer: General purpose the key and value  components that have to be put in. getorDefault is the output value to the corresponding key and if there is none it can simply be returned so the general purpose is required 
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: .map is a higher order function that iterates through every element in an array, while .filter is a high order function that loops through every element. 
 
-  Researched answer:
+  Researched answer: Both are higher order functions .map affects every element in the array and returns different values but the same length, .filter affects different elements of the array based on the guidelines given and length differs.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: Methods are built-in and are actions used on objects
```

Comment:
Methods can be custom as well, however the key difference between a function and a method is that a method belongs to an object.